Allow a custom poll time to be set

mattsoftware opened the pull request:
Added the ability to set poll time, rather than the hardcoded 2000ms value. It still defaults to 2000 to not break existing behaviour.

There is a guard against the number being used. I've set a lower limit of one second (there was a comment to say it could be bumped lower than two at the cost of cpu). Perhaps that could be put up to two seconds, but in my opinion if someone wanted to sacrifice cpu for a lower poll time that would be up to them. I've also set an upper limit of 60 seconds. That's arbitrary and I didn't really know what to set it to. My thinking was if they wanted longer than 60 seconds they would likely want to put the device into a low power state and wake it up at the time they want a new coordinate.

I have made the variable private so to enforce the limit's of the value and stop accidental modification of the number into something that will not work.

jiahuang commented on index.js:
```diff
@@ -93,6 +93,23 @@ var GPS = function (hardware, callback) {
+  self.setPollTime = function (time) {
+    time = parseInt(time, 10);
+    if (time < 2000) {
+        // No less than 2 seconds
```
Can you add a console.log that just spits out something like "GPS Poll time cannot be set less than 2 seconds, defaulting to 2 seconds."?

Just don't want the user confused about why it's not polling constantly if they set a low value.

jiahuang commented on index.js:
```diff
@@ -93,6 +93,23 @@ var GPS = function (hardware, callback) {
+        // No less than 2 seconds
+        time = 2000;
+    } else if (time > 60000) {
+        // No more than 60 seconds
```
Same thing, a console.log about the maximum poll time would be helpful here.

jiahuang commented:
Thanks for the PR! I added a few comments mainly about exposing up to the user what values are being defaulted to. Other than that, lgtm.

mattsoftware commented:
I've updated with some console.log messages. I'm only doing that when in debug mode. I've also found a few test cases I was not working well with, so i've fixed them too. I have a test file but it always fails for some reason, i might commit it on a different branch to get some feedback on the test file

tm-rampart commented:
Approved by @jiahuang. [Running tests](https://rampart.tessel.io/jobs/6023).
